qualtran/_infra/gate_with_registers.py

```diff
@@ -149,16 +148,37 @@ class GateWithRegisters(Bloq, cirq.Gate, metaclass=abc.ABCMeta):
     def decompose_bloq(self) -> 'CompositeBloq':
         """Decompose this Bloq into its constituent parts contained in a CompositeBloq.
 
-        Bloq users can call this function to delve into the definition of a Bloq. If you're
-        trying to define a bloq's decomposition, consider overriding `build_composite_bloq`
-        which provides helpful arguments for implementers.
+        Bloq users can call this function to delve into the definition of a Bloq. The function
+        returns the decomposition of this Bloq represented as an explicit compute graph wrapped
+        in a `CompositeBloq` object.
+
+        Bloq authors can the bloq's decomposition by overriding any of the two convenience methods:
+
+        - `build_composite_bloq`: Override this method to define a bloq-style decomposition using a
+            `BloqBuilder` builder class to construct the `CompositeBloq` directly.
+        - `decompose_from_registers`: Override this method to define a cirq-style decomposition by
+            yielding cirq style operations applied on qubits.
+
+        Irrespective of the choice of backend chosen by the Bloq authors to implement the
+        decomposition, Bloq users will be able to access both the Bloq-style and Cirq-style
+        interfaces, i.e. for a bloq that inherits from `GateWithRegisters`, usrs can do:
+
+        - `cirq.decompose_once(bloq.on_registers(**cirq_quregs))`: This will yield a cirq-style
+            `OPTREE` where only bloqs which do not inherit from `GateWithRegisters` will be wrapped
+            in a `BloqAsCirqGate` wrapper.
+        - `bloq.decompose_bloq()`: This will return a `CompositeBloq` where only cirq gates which do
+            not inherit from `GateWithRegisters` will be be wrapped in a `CirqGateAsBloq` wrapper.
```

Comment:
```suggestion
        - `bloq.decompose_bloq()`: This will return a `CompositeBloq`.
           Cirq gates will be be wrapped in `CirqGateAsBloq` as needed.
```